doc/md/5-actors-async.md

```diff
@@ -0,0 +1,316 @@
+
+- Each call sends an extra message to the actor.
+
+- Every call must be awaited, adding overhead.
+
+- Each `await` suspends execution, increasing chances of interference from other concurrent messages.
+
+To reduce the overhead and risks of extra `await`s, Motoko provides computation types, written as `async* T`. Like futures (`async T`), computations can represent asynchronous tasks.
+
+An `async` expression creates a future by starting its execution immediately, while an `async*` expression creates a computation by delaying execution until needed. Similarly, `await` gets the result of a future, and `await*` gets the result of a computation by triggering its next step.
+
+From a typing perspective, futures and computations are similar, but they behave differently at runtime. A future is a stateful object representing a scheduled asynchronous task, while a computation is an inactive value that describes a task.
+
+When you use `await` on a future, it suspends the caller until the task finishes. But `await*` on a computation doesn’t suspend the caller; it immediately runs the computation like a normal function call.
+
+This means that `await*` only causes suspension if the computation’s body itself uses a regular `await`. The `*` indicates that the computation might include zero or more `await` calls and so may be interleaved with other message executions.
+
+You create an `async*` value by using an `async*` expression, but usually, it’s done by defining a local function that returns an `async*` type.
+
+To get the result of an `async*` computation, you use `await*`.
+
+:::danger
+
+Use `async*` and `await*` carefully. In Motoko, a regular `await` is a commit point. State changes are saved before the function pauses.
+
+`await*` is not a commit point because the computation it runs may not pause or commit at a predictable time. This means if a trap happens inside an `await*` computation, the actor’s state will roll back to the last commit point before the `await*`, not to the point of the `await*` itself.
+
+:::
+
+### Example
+
+Customers can place multiple orders at the same time. However, when they ask for an update on their delivery status, responses must be generated sequentially based on the preparation time. Instead of constantly tracking delivery updates, `async*` can be used to calculate the estimated delivery time only when requested.
+
+```motoko no-repl
+// async* function to calculate delivery time on demand
+func checkDelivery(order : Text) : async* Text {
+        let estimatedTime = 15 + (orders.size() * 5); // Base time + 5 mins per pending order
+        return "Your " # order # " will arrive in " # Nat.toText(estimatedTime) # " minutes.";
+    };
+
+    // Multiple users can check their delivery status asynchronously
+    public shared func getDeliveryStatus(order : Text) : async Text {
+        let status = await* checkDelivery(order);
+        return status;
+    };
+```
+
+## `try/finally`
```

Comment:
Are try-catch and try-catch-finally explained anywhere? There's used to be section on this in error handling but it seems to have disappeared.